app/src/main/java/org/mozilla/fenix/library/downloads/DownloadView.kt

```diff
@@ -55,18 +73,44 @@ class DownloadView(
     }
 
     fun update(state: DownloadFragmentState) {
+        val oldMode = mode
 
+        view.progress_bar.isVisible = state.isDeletingItems
         view.swipe_refresh.isEnabled = false
         mode = state.mode
 
-        updateEmptyState(state.items.isNotEmpty())
+        downloadAdapter.updatePendingDeletionIds(state.pendingDeletionIds)
+
+        updateEmptyState(state.pendingDeletionIds.size != state.items.size)
 
         downloadAdapter.updateMode(state.mode)
         downloadAdapter.updateDownloads(state.items)
 
-        setUiForNormalMode(
-            context.getString(R.string.library_downloads)
-        )
+        if (state.mode::class != oldMode::class) {
+            interactor.onModeSwitched()
+        }
+
+        if (state.mode is DownloadFragmentState.Mode.Editing) {
+            val unselectedItems = oldMode.selectedItems - state.mode.selectedItems
+
+            state.mode.selectedItems.union(unselectedItems).forEach { item ->
+                val index = state.items.indexOf(item)
+                downloadAdapter.notifyItemChanged(index)
+            }
+        }
```

Comment:
Move this down into the `when` (line 108)

Reply:
@kglazko I think this change is still pending :)